compiler/parser/src/function.rs

```diff
@@ -54,7 +53,8 @@ pub fn parse_args(func_args: Vec<FunctionArgument>) -> Result<ArgumentList, Lexi
     let mut args = vec![];
     let mut keywords = vec![];
 
-    let mut keyword_names = HashSet::with_capacity_and_hasher(func_args.len(), RandomState::new());
+    let mut keyword_names =
+        FnvHashSet::with_capacity_and_hasher(func_args.len(), Default::default());
```

Comment:
does using `fnv` make such a difference? Reading on it the benefit seems to be small inputs (of which keyword args commonly are probably). As an aside, [rustc-hash](https://crates.io/crates/rustc-hash) is another option (and supposedly faster).

Reply:
This change alone (that is, excluding the changes in `string.rs`) drops execution time to ~263ms from ~268ms. So the two changes contribute close to equally, at least in my benchmark.

`rustc-hash` seems to be a tiny bit faster, even:

```
Benchmark 1: ./target/release/ruff resources/test/cpython --no-cache
  Time (mean ± σ):     254.2 ms ±   3.8 ms    [User: 2149.0 ms, System: 82.6 ms]
  Range (min … max):   248.8 ms … 266.1 ms    50 runs

Benchmark 1: ./target/release/ruff resources/test/cpython --no-cache
  Time (mean ± σ):     254.9 ms ±   4.8 ms    [User: 2164.6 ms, System: 82.4 ms]
  Range (min … max):   245.5 ms … 266.9 ms    50 runs

Benchmark 1: ./target/release/ruff resources/test/cpython --no-cache
  Time (mean ± σ):     254.8 ms ±   4.6 ms    [User: 2159.1 ms, System: 83.1 ms]
  Range (min … max):   246.5 ms … 269.4 ms    50 runs
```